### DISCUSSION TITLE
Filter only facets not appearing.

### DISCUSSION THREAD

**Discussion by mikebhart** [algolia/instantsearch#4488] [General] [answered]
Hi there I have setup attributes for faceting in the Configuration | Facets section.  It shows for non searchable attributes but doesnt show for filter only.

**Comment by Haroenv** [comment, marked as answer]
FilterOnly is exactly that, you can filter them, but not list the values. Therefore a refinementlist doesn’t work on a filterOnly facet.

**Reply by mikebhart** [reply]
Thank you for your help much appreciated, I don't understand why. If you cant see the values how can you filter them?.  What I want is a dropdown select box to work in the same way as refinmentlist checkboxes.  Do you know how I would achieve this, many thanks in advance?

**Reply by Haroenv** [reply]
FilterOnly is an optimisation for attributes for which have many values, upon which you want to filter, but not show its values in a UI (just apply the facet for business rules) and would take less memory usage for the server. In the general case this isn't needed, and you can have `attributesForFaceting: ['myFacetName']`

A dropdown select box is `menuSelect`: https://www.algolia.com/doc/api-reference/widgets/menu-select/js

**Reply by mikebhart** [reply]
thanks again.  Do you know where I can get the code for this example demo https://instantsearchjs.netlify.app/stories/?path=/story/refinements-menuselect--default

also with the demos examples is there a way of logging into the alogliasearch dashboard to see how the data is constructed.

**Reply by Haroenv** [reply]
the code is here: https://github.com/algolia/instantsearch.js/blob/master/stories/menu-select.stories.js

the demo we use for all `instant_search` has records like this:

```json
{
  "name": "Amazon - Fire TV Stick with Alexa Voice Remote - Black",
  "description": "Enjoy smart access to videos, games and apps with this Amazon Fire TV stick. Its Alexa voice remote lets you deliver hands-free commands when you want to watch television or engage with other applications. With a quad-core processor, 1GB internal memory and 8GB of storage, this portable Amazon Fire TV stick works fast for buffer-free streaming.",
  "brand": "Amazon",
  "categories": [
    "TV & Home Theater",
    "Streaming Media Players"
  ],
  "hierarchicalCategories": {
    "lvl0": "TV & Home Theater",
    "lvl1": "TV & Home Theater > Streaming Media Players"
  },
  "type": "Streaming media plyr",
  "price": 39.99,
  "price_range": "1 - 50",
  "image": "https://cdn-demo.algolia.com/bestbuy-0118/5477500_sb.jpg",
  "url": "https://api.bestbuy.com/click/-/5477500/pdp",
  "free_shipping": false,
  "rating": 4,
  "popularity": 21469,
  "objectID": "5477500"
}
```

with these settings:

```json
{
  "minWordSizefor1Typo": 3,
  "minWordSizefor2Typos": 7,
  "hitsPerPage": 20,
  "maxValuesPerFacet": 100,
  "version": 2,
  "searchableAttributes": [
    "unordered(categories)",
    "unordered(type)",
    "brand",
    "unordered(name)",
    "unordered(description)"
  ],
  "numericAttributesToIndex": null,
  "attributesToRetrieve": null,
  "allowTyposOnNumericTokens": true,
  "ignorePlurals": true,
  "advancedSyntax": false,
  "removeStopWords": false,
  "replaceSynonymsInHighlight": true,
  "distinct": false,
  "unretrievableAttributes": null,
  "optionalWords": null,
  "slaves": [
    "instant_search_price_desc",
    "instant_search_price_asc",
    "instant_search_rating_asc"
  ],
  "attributesForFaceting": [
    "searchable(brand)",
    "price_range",
    "categories",
    "type",
    "price",
    "free_shipping",
    "hierarchicalCategories.lvl0",
    "hierarchicalCategories.lvl1",
    "hierarchicalCategories.lvl2",
    "hierarchicalCategories.lvl3",
    "hierarchicalCategories.lvl4",
    "rating",
    "popularity"
  ],
  "attributesToSnippet": null,
  "attributesToHighlight": null,
  "paginationLimitedTo": 1000,
  "attributeForDistinct": null,
  "exactOnSingleWordQuery": "attribute",
  "ranking": [
    "typo",
    "geo",
    "words",
    "filters",
    "proximity",
    "attribute",
    "exact",
    "custom"
  ],
  "customRanking": [
    "desc(popularity)"
  ],
  "separatorsToIndex": "",
  "removeWordsIfNoResults": "lastWords",
  "queryType": "prefixLast",
  "highlightPreTag": "<em>",
  "highlightPostTag": "</em>",
  "snippetEllipsisText": "",
  "typoTolerance": "true",
  "alternativesAsExact": [
    "ignorePlurals",
    "singleWordSynonym"
  ],
  "sortFacetValuesBy": "count"
}
```

**Comment by mikebhart** [comment]
I tried to install the demo  but get the following when installing

C:\dev\code\instantsearch.js-master>npm install
npm WARN deprecated rollup-plugin-babel@4.3.3: This package has been deprecated and is no longer maintained. Please use @rollup/plugin-babel.
npm WARN deprecated rollup-plugin-commonjs@10.1.0: This package has been deprecated and is no longer maintained. Please use @rollup/plugin-commonjs.
npm WARN deprecated rollup-plugin-node-resolve@5.2.0: This package has been deprecated and is no longer maintained. Please use @rollup/plugin-node-resolve.
npm WARN deprecated rollup-plugin-replace@2.2.0: This module has moved and is now available at @rollup/plugin-replace. Please update your dependencies. This version is no longer maintained.
npm WARN deprecated mkdirp@0.3.0: Legacy versions of mkdirp are no longer supported. Please update to mkdirp 1.x. (Note that the API surface has changed to use Promises in 1.x.)
npm WARN deprecated fsevents@1.2.13: fsevents 1 will break on node v14+ and could be using insecure binaries. Upgrade to fsevents 2.
npm WARN deprecated popper.js@1.16.1: You can find the new Popper v2 at @popperjs/core, this package is dedicated to the legacy v1
npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
npm WARN deprecated iltorb@2.4.5: The zlib module provides APIs for brotli compression/decompression starting with Node.js v10.16.0, please use it over iltorb
npm WARN deprecated request-promise-native@1.0.9: request-promise-native has been deprecated because it extends the now deprecated request package, see https://github.com/request/request/issues/3142
npm WARN deprecated har-validator@5.1.5: this library is no longer supported
npm ERR! code 1
npm ERR! Command failed: git checkout 1.3.0
npm ERR! error: pathspec '1.3.0' did not match any file(s) known to git
npm ERR! 


I just want a menuselect example on codepen or something with a fully working example.

**Reply by Haroenv** [reply]
https://codesandbox.io/s/stupefied-bird-b7lv1
